client/blocks/inline-help/inline-help-search-card.jsx

```diff
@@ -69,40 +65,39 @@ class InlineHelpSearchCard extends Component {
 		}
 	};
 
-	searchHelperHandler = ( searchQuery ) => {
-		const query = searchQuery.trim();
-		if ( ! query || ! query.length ) {
-			return debug( 'empty query. Skip recording tracks-event.' );
-		}
-
-		debug( 'search query received: ', searchQuery );
-		this.props.recordTracksEvent( 'calypso_inlinehelp_search', {
-			search_query: searchQuery,
-			location: 'inline-help-popover',
-		} );
-
-		// Make a search
-		this.props.requestInlineHelpSearchResults( searchQuery );
-	};
-
 	componentDidMount() {
-		this.props.requestInlineHelpSearchResults();
+		this.props.requestInlineSearchResultsAndTrack();
 	}
 
 	render() {
 		return (
 			<SearchCard
 				searching={ this.props.isSearching }
 				initialValue={ this.props.query }
-				onSearch={ this.searchHelperHandler }
+				onSearch={ this.props.requestInlineSearchResultsAndTrack }
 				onKeyDown={ this.onKeyDown }
 				placeholder={ this.props.translate( 'Search for help…' ) }
 				delaySearch={ true }
+				onSearchClose={ this.props.resetSearchResults }
 			/>
 		);
 	}
 }
 
+const requestInlineSearchResultsAndTrack = ( searchQuery ) => (
+	! searchQuery || ! ( searchQuery.trim() ).length
```

Comment:
Could we reverse that logic and make it a positive check to make it a little easier to understand what's happening here?

Reply:
Agreed. I've reversed in _both_ locations.